gateway/api/services/storage/logs_storage.py

```diff
@@ -0,0 +1,92 @@
+"""
+This module handle the access to the logs store
+"""
+import logging
+import os
+from typing import Optional
+
+from api.services.storage.enums.working_dir import WorkingDir
+from api.services.storage.path_builder import PathBuilder
+
+
+logger = logging.getLogger("gateway")
+
+
+class LogsStorage:
+    """
+    The main objective of this class is to manage the access to logs generated by a Job.
+
+    Attributes:
+        username (str): storage user's username
+        working_dir (WorkingDir(Enum)): working directory
+        function_title (str): title of the function in case is needed to build the path
+        provider_name (str | None): name of the provider in caseis needed to build the path
+    """
+
+    FILE_EXTENSION = ".log"
+    PATH = "logs"
+    ENCODING = "utf-8"
+
+    def __init__(
+        self,
+        username: str,
+        working_dir: WorkingDir,
+        function_title: str,
+        provider_name: Optional[str],
+    ) -> None:
+        self.sub_path = PathBuilder.get_sub_path(
+            working_dir=working_dir,
+            username=username,
+            function_title=function_title,
+            provider_name=provider_name,
+            extra_sub_path=self.PATH,
+        )
+        self.absolute_path = PathBuilder.get_absolute_path(
+            working_dir=working_dir,
+            username=username,
+            function_title=function_title,
+            provider_name=provider_name,
+            extra_sub_path=self.PATH,
+        )
+
+    def _get_log_path(self, job_id: str) -> str:
```

Comment:
```suggestion
    def _get_logs_path(self, job_id: str) -> str:
```
I think logs it is usually plural always because you use to have multiple log lines.